### DISCUSSION TITLE
[Help]: Is it possible to preload the slides in the carousel?

### DISCUSSION THREAD

**Discussion by steve-marmalade** [davidjerleke/embla-carousel#695] [General]
Hello, I am new to embla and really enjoying it so far.

One thing I've noticed is that when scrolling through the carousel there is a brief period of time where the blurUrl for my images are shown before the full res version displays. My suspicion is that embla is lazily loading the images (which makes sense as a default!). I was able to test this by manually looping through the images and preload them, and indeed the delay goes away.

Am I correct about how embla works? Is it currently possible to tell embla to preload the slides in the carousel? If not, this might be a useful feature (especially if it was still lazy by default, but then preloaded upon an interaction like a touch or hover event).

**Comment by davidjerleke** [comment]
> Am I correct about how embla works? Is it currently possible to tell embla to preload the slides in the carousel? If not, this might be a useful feature (especially if it was still lazy by default, but then preloaded upon an interaction like a touch or hover event).

@steve-marmalade thank you for your question. No, **Embla doesn't touch any images** or anything else inside the slides for that matter. The only two things Embla does are the following:

1. It applies `translate3d` to the **container node** in order to create the scroll effect.
2. It applies `translate3d` to some **slide nodes** in order to create the loop effect when **loop** is **enabled**. 

Nothing else actually. So the behaviour you're seeing might be related to the browser you are using. You didn't share any information about what browser or tools you're using so I can't say what's causing this behaviour.

Best,
David

**Reply by davidjerleke** [reply]
@steve-marmalade did you see [my response](https://github.com/davidjerleke/embla-carousel/discussions/695#discussioncomment-8140842)?
